chain/pool/txpool.go

```diff
@@ -243,6 +262,11 @@ func (tp *TxnPool) AppendTxnPool(txn *transaction.Transaction) error {
 		return err
 	}
 
+	// 5. sync txn, add to txn with time list
+	if !bExist { // append it into list only when it doesn't exist in the pool before.
```

Comment:
Yes. Here mainly is to avoid sync in the txn which the node just sync out. 
I modify it to :
`
// sync txn, we trace if this txn is exsit this pool
var oldTxn *transaction.Transaction
if oldTxn, err = list.GetByNonce(txn.UnsignedTx.Nonce); err != nil && list.Full() {
	return errors.New("account txpool full, too many transaction in list")
}

...

// 5. sync txn, add to txn with time map
// append it into list only when it doesn't exist in the pool before or hash is different
if oldTxn == nil || bytes.Compare(oldTxn.Hash(), txn.Hash()) != 0 {
	tp.AppendTxnWithTime(txn)
}
`

Reply:
It's already checked before calling AppengTxnPool https://github.com/nknorg/nkn/blob/master/lnode/transaction.go#L375